tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,222 @@
+	resourcecondition "maistra.io/istio-operator/pkg/util/tests/types"
+)
+
+var _ = Describe("Operator", Ordered, func() {
+	SetDefaultEventuallyTimeout(120 * time.Second)
+	SetDefaultEventuallyPollingInterval(time.Second)
+	var resourceAvailable, resourceReconcilied, resourceReady resourcecondition.Conditions
+
+	Describe("operator installation", func() {
+		// TODO: we  need to support testing both types of deployment for the operator, helm and olm via subscription.
+		// Discuss with the team if we should add a flag to the test to enable the olm deployment and don't do that deployment in different step
+		When("default helm manifest are applied", func() {
+			BeforeEach(func() {
+				resourceAvailable = resourcecondition.Conditions{
+					Type:   "Available",
+					Status: "True",
+				}
+
+				if skipDeploy == "true" {
+					Skip("Skipping the deployment of the operator and the tests")
+				}
+				GinkgoWriter.Println("Deploying Operator using default helm charts located in /chart folder")
+				Eventually(deployOperator).Should(Succeed(), "Operator deployment should be successful")
+			})
+
+			Specify("the operator is running", func() {
+				Eventually(func(g Gomega) {
+					g.Expect(kubectl.GetResourceCondition(namespace, "deployment", deploymentName)).Should(ContainElement(resourceAvailable))
+				}).Should(Succeed(), "Operator deployment should be Available")
+				GinkgoWriter.Println("Operator deployment is Available")
+
+				podName, err := kubectl.GetPodFromLabel(namespace, "control-plane=istio-operator")
+				GinkgoWriter.Println("Istio Operator pod name: ", podName)
+				if err != nil {
+					Fail("Error getting pods from label")
+				}
+
+				Eventually(kubectl.GetPodPhase).WithArguments(namespace, podName).Should(Equal("Running"), "Istio-operator pod should be Running")
+				GinkgoWriter.Println("Istio-operator pod is Running")
+			})
+		})
+	})
+
+	Describe("installation and unistallation of the istio resource", func() {
+		for _, version := range istioVersions {
+			// Note: This var version is needed to avoid the closure of the loop
+			version := version
+
+			Context(fmt.Sprintf("is applied the istio resource with version %s", version), func() {
+				BeforeAll(func() {
+					Expect(kubectl.CreateNamespace(controlPlaneNamespace)).To(Succeed())
+					Eventually(createIstioCR).WithArguments(version).Should(Succeed(), "Istio CR should be created")
+					GinkgoWriter.Println("Istio CR created")
+
+					resourceReconcilied = resourcecondition.Conditions{
+						Type:   "Reconciled",
+						Status: "True",
+					}
+
+					resourceReady = resourcecondition.Conditions{
+						Type:   "Ready",
+						Status: "True",
+					}
+				})
+
+				AfterAll(func() {
+					Eventually(kubectl.DeleteNamespace).WithArguments(controlPlaneNamespace).Should(Succeed(), "Namespace should be deleted")
+					Eventually(kubectl.GetNamespace).WithArguments(controlPlaneNamespace).Should(Equal(fmt.Sprintf("namespace %s not found", controlPlaneNamespace)))
```

Comment:
Done

Reply:
I had to change the func to CHeckNamespaceExist, MatchError does not work as we expect because it's not checking the err returned by the func it's checking that the first argument returns an error and this condition is not ever matched. I'll create a todo for this to improve this